proxy/proxy.go

```diff
@@ -469,6 +469,8 @@ func setRequestURLForDynamicBackend(u *url.URL, stateBag map[string]interface{})
 }
 
 func setRequestURLForLoadBalancedBackend(u *url.URL, rt *routing.Route, lbctx *routing.LBContext) *routing.LBEndpoint {
+	lbctx.LBEndpoints = rt.LBEndpoints
```

Comment:
Lets do this either at the place where context is created or maybe better move context creation here.

Reply:
> Lets do this either at the place where context is 
I like this idea, I've done that.

> or maybe better move context creation here.
I like this way less, because I need to redeclare a lot variables from `func mapRequest(...` in `func setRequestURLForLoadBalancedBackend(...`
```
diff --git a/proxy/proxy.go b/proxy/proxy.go
index 80b41df3..dddf641f 100644
--- a/proxy/proxy.go
+++ b/proxy/proxy.go
@@ -468,7 +468,13 @@ func setRequestURLForDynamicBackend(u *url.URL, stateBag map[string]interface{})
        }
 }
 
-func setRequestURLForLoadBalancedBackend(u *url.URL, rt *routing.Route, lbctx *routing.LBContext) *routing.LBEndpoint {
+func setRequestURLForLoadBalancedBackend(ctx *context, registry *routing.EndpointRegistry) *routing.LBEndpoint {
+       r := ctx.request
+       rt := ctx.route
+       stateBag := ctx.StateBag()
+       u := r.URL
+       lbctx := &routing.LBContext{Request: r, Route: rt, LBEndpoints: rt.LBEndpoints, Params: stateBag, Registry: registry}
+
        e := rt.LBAlgorithm.Apply(lbctx)
        u.Scheme = e.Scheme
        u.Host = e.Host
@@ -490,7 +496,7 @@ func mapRequest(ctx *context, requestContext stdlibcontext.Context, removeHopHea
                setRequestURLFromRequest(u, r)
                setRequestURLForDynamicBackend(u, stateBag)
        case eskip.LBBackend:
-               endpoint = setRequestURLForLoadBalancedBackend(u, rt, &routing.LBContext{Request: r, Route: rt, LBEndpoints: rt.LBEndpoints, Params: stateBag, Registry: registry})
+               endpoint = setRequestURLForLoadBalancedBackend(ctx, registry)
        default:
                u.Scheme = rt.Scheme
                u.Host = rt.Host

```

Reply:
This is actually better because it nicely contains all endpoint selection logic and creates a place to extend in the future.
Also  
```
	u.Scheme = e.Scheme
	u.Host = e.Host
```
can be moved out of setRequestURLForLoadBalancedBackend as its not related to endpoint selection,
setRequestURLForLoadBalancedBackend could be renamed to selectLoadBalancedEndpoint (or something alike)

Reply:
https://github.com/zalando/skipper/pull/2652